pkg/server/v3/delta_test.go

```diff
@@ -0,0 +1,413 @@
+package server_test
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"reflect"
+	"testing"
+	"time"
+
+	"google.golang.org/grpc"
+
+	discovery "github.com/envoyproxy/go-control-plane/envoy/service/discovery/v3"
+	"github.com/envoyproxy/go-control-plane/pkg/cache/types"
+	"github.com/envoyproxy/go-control-plane/pkg/cache/v3"
+	rsrc "github.com/envoyproxy/go-control-plane/pkg/resource/v3"
+	"github.com/envoyproxy/go-control-plane/pkg/server/stream/v3"
+	"github.com/envoyproxy/go-control-plane/pkg/server/v3"
+)
+
+func (config *mockConfigWatcher) CreateDeltaWatch(req *discovery.DeltaDiscoveryRequest, st *stream.StreamState) (chan cache.DeltaResponse, func()) {
+	config.deltaCounts[req.TypeUrl] = config.deltaCounts[req.TypeUrl] + 1
+
+	// Create our out watch channel to return with a buffer of one
+	out := make(chan cache.DeltaResponse, 1)
+
+	if len(config.deltaResponses[req.TypeUrl]) > 0 {
+		res := config.deltaResponses[req.TypeUrl][0]
+		// In subscribed, we only want to send back what's changed if we detect changes
+		var subscribed []types.Resource
+
+		// Here we set our current version map to what the stream state says it is
+		versionMap := st.ResourceVersions
+
+		r, _ := res.GetDeltaDiscoveryResponse()
+		if st.IsWildcard {
+			for _, resource := range r.Resources {
+				name := resource.GetName()
+				marshaledResource, _ := cache.MarshalResource(resource)
+				oldV := versionMap[name]
+				if v := cache.HashResource(marshaledResource); v != oldV {
+					versionMap[name] = v
+				}
+				subscribed = append(subscribed, resource)
+			}
+		} else {
+			if len(req.GetResourceNamesSubscribe()) > 0 {
+				for _, resource := range r.Resources {
+					for _, alias := range req.GetResourceNamesSubscribe() {
+						if name := resource.GetName(); name == alias {
+							marshaledResource, _ := cache.MarshalResource(resource)
```

Comment:
maybe panic at the very least here instead of dropping `err` so we'd surface the error?